make check//unix/-fPIE/-pie RUNTESTFLAGS=solib-display.exp

gcc -fpic -c  -fPIE -pie -o x.o x.c
/usr/lib/gcc/x86_64-redhat-linux/4.6.1/../../../../lib64/Scrt1.o: In function `_start':
(.text+0x20): undefined reference to `main'

=> Change the order for overrides.

One has to also use -fPIC rather than -fPIE, -fPIC is stronger.

The correct way would be:
make check//unix RUNTESTFLAGS='CC_FOR_TARGET=gcc\ -fPIC\ -pie CXX_FOR_TARGET=g++\ -fPIC\ -pie solib-display.exp'

But there is a problem with testsuite.unix non-unique subdir name and also
a problem with make -j parallelization of the testsuite.

--- gdb-7.3.50.20110722/gdb/testsuite/lib/future.exp	2011-10-11 16:44:05.000000000 +0200
+++ gdb-7.3.50.20110722/gdb/testsuite/lib/future.exp	2011-10-11 16:44:10.000000000 +0200
@@ -77,6 +77,10 @@ proc gdb_default_target_compile {source
     set ldflags ""
     set dest [target_info name]
 
+    if {[board_info $dest exists multilib_flags]} {
+	append add_flags " [board_info $dest multilib_flags]"
+    }
+
     if {[info exists CFLAGS_FOR_TARGET]} {
 	append add_flags " $CFLAGS_FOR_TARGET"
     }
@@ -333,10 +337,6 @@ proc gdb_default_target_compile {source
 	}
     }
 
-    if {[board_info $dest exists multilib_flags]} {
-	append add_flags " [board_info $dest multilib_flags]"
-    }
-
     verbose "doing compile"
 
     set sources ""
